### DISCUSSION TITLE
minikube start and cannot got metrics of etcd

### DISCUSSION THREAD

**Discussion by reply2future** [prometheus-operator/kube-prometheus#1760] [Q&A]
I used [the command](https://github.com/prometheus-operator/kube-prometheus#minikube) to setup minikube. Then I install the [kube-prometheus-stack](https://artifacthub.io/packages/helm/prometheus-community/kube-prometheus-stack) using helm chart. After that I saw a error alert firing because cannot receive the metrics from the etcd, I checked the target status like this:
![image](https://user-images.githubusercontent.com/6359152/168563689-86b798ae-4623-4a5d-a450-5d050f9e49fb.png)

I checked the settings of `etcd-minikube` like this:
![image](https://user-images.githubusercontent.com/6359152/168563952-6f397cff-4cff-4803-8015-234ec66c41c7.png)
And I used command `curl -v 'http://0.0.0.0:2381/metrics'` and displayed the metrics information after using `minikube ssh`

I also tried to change the port of `etcd service`( kube-prometheus-stack-kube-etcd ) to `2381` but not working.

Appreciate any help.

**Comment by perrness** [comment]
Try adding `--extra-config=etcd.listen-metrics-urls=http://0.0.0.0:2381` to `minikube start`.
For anyone else stumbling across this issue, you also need to change the port for the etcd service. Add this to your `values.yaml` when installing/updating the helm chart.
```yaml
kubeEtcd:
  service:
    enabled: true
    port: 2381
    targetPort: 2381
```

**Reply by reply2future** [reply]
Thanks for your reply.But the key issue is that the metric port of prometheus to check is `2379` not `2381`.
